crates/api/src/chronicle_graphql/mod.rs

```diff
@@ -366,6 +377,49 @@ async fn gql_playground() -> impl IntoResponse {
     ))
 }
 
+#[derive(Clone, Debug)]
+pub struct JwtClaims(pub serde_json::Map<String, serde_json::Value>);
+
+struct AuthorizationEndpoint<Q, M, S> {
+    checker: JwtChecker,
+    schema: Schema<Q, M, S>,
+}
+
+#[poem::async_trait]
+impl<Q, M, S> Endpoint for AuthorizationEndpoint<Q, M, S>
+where
+    Q: ObjectType + 'static,
+    M: ObjectType + 'static,
+    S: SubscriptionType + 'static,
+{
+    type Output = GraphQLBatchResponse;
+
+    async fn call(&self, req: poem::Request) -> poem::Result<Self::Output> {
+        if let Some(authorization) = req.header("Authorization") {
+            let authorization = HeaderValue::from_str(authorization).unwrap();
```

Comment:
No unwraps in this sort of code please, that's an easy DOS attack

Reply:
Can we span / instrument this section so it's easy to get timings out of it. Either use the span!().in_scope(|| or extract to another fn and #[insurement] it

Reply:
This refers to the verify_jwt bit, we should get decent enough spans once the request is executng

Reply:
Thanks, was worth the rewrite to eliminate `unwrap()`.